nixos/modules/tasks/network-interfaces.nix

```
@@ -1058,10 +1064,12 @@ in
           };
 
           type = mkOption {
-            type = with types; enum [ "tun" "tap" ];
+            type = with types; enum [ "tun" "tap" "tun6" "tap6" ];
             default = "tap";
             example = "tap";
-            apply = v: if v == "tun" then "gre" else "gretap";
+            apply = v: if stringLength v == 3
+              then (if v == "tun" then "gre" else "gretap")
+              else (if v == "tun6" then "ip6gre" else "ip6gretap");
```

```suggestion
            apply = v: {
              tun = "gre";
              tap = "gretap";
              tun6 = "ip6gre";
              tap6 = "ip6gretap";
            }.${v};
```
this is a little more verbose, but also a little more robust should more flags be added to the enum in the future.

---

I have locally tested the new test and applied your suggestion.